src/supautils.c

```diff
@@ -432,6 +536,93 @@ supautils_hook(PROCESS_UTILITY_PARAMS)
 			break;
 		}
 
+		case T_CommentStmt:
+		{
+			if (!IsTransactionState()) {
+				break;
+			}
+			if (superuser()) {
+				break;
+			}
+			if (((CommentStmt *)utility_stmt)->objtype != OBJECT_EXTENSION) {
+				break;
+			}
+			if (privileged_role == NULL) {
+				break;
+			}
+			if (!OidIsValid(get_role_oid(privileged_role, true))) {
+				break;
+			}
+			if (GetUserId() != get_role_oid(privileged_role, false)) {
+				break;
+			}
+
+			{
+				Oid superuser_oid = BOOTSTRAP_SUPERUSERID;
+				Oid prev_role_oid;
+				int prev_role_sec_context;
+
+				if (privileged_extensions_superuser != NULL) {
+					superuser_oid = get_role_oid(privileged_extensions_superuser, false);
+				}
+
+				GetUserIdAndSecContext(&prev_role_oid, &prev_role_sec_context);
+				SetUserIdAndSecContext(superuser_oid, prev_role_sec_context |
+									   SECURITY_LOCAL_USERID_CHANGE |
+									   SECURITY_RESTRICTED_OPERATION);
+
+				run_process_utility_hook(prev_hook);
+
+				SetUserIdAndSecContext(prev_role_oid, prev_role_sec_context);
+
+				return;
+			}
+		}
+
+		case T_VariableSetStmt:
+		{
+			if (!IsTransactionState()) {
+				break;
+			}
+			if (superuser()) {
+				break;
+			}
+			// TODO: refactor to support other variables
+			if (strcmp(((VariableSetStmt *)utility_stmt)->name, "session_replication_role") != 0) {
```

Comment:
Hm, having a config for GUCs would be useful for the postgREST in-db config as well. 

The following requires superuser and will break once it's gone

```
ALTER ROLE authenticator SET pgrst.db_pre_request TO '..';
```

Reply:
How about making it configurable now? Otherwise later it'd be a breaking change

Reply:
It would be even be very useful to allow a `prefix` for the GUCs allowed. That way we don't have to whitelist every postgREST config and just do `supautils.configurable_guc = 'pgrst.*'`.

Reply:
Break as in, once this PR is applied non-superusers are allowed to do the `alter role`?

Reply:
Or do you mean once we change the dashboard role to non-superuser, users can't do the `alter role`?

Reply:
(I'm assuming the latter - will make it a config)

Reply:
> Or do you mean once we change the dashboard role to non-superuser, users can't do the alter role?

Once superuser is removed, users will not be able to do ALTER ROLE on placeholders: `ALTER ROLE <any_role> SET any_prefix.conf TO '..'`

Reply:
Updated the PR, but I just realized it still doesn't work on the `alter role` above since `authenticator` is a reserved role.

Let's just punt these for now since these features aren't trivial (e.g. should the privileged role be able to `alter role` on superusers?)

Reply:
Yeah, agree. The `privileged_role_allowed_configs` seems like a good start to later solve the `authenticator` issue.